src/lib.rs

```diff
@@ -159,6 +204,21 @@ fn test_pop_push() {
     assert_eq!([1], *a);
 }
 
+#[test]
+fn test_update() {
```

Comment:
What should the behavior be if the user tries an out-of-bounds index? I don't think a panic is unreasonable as long as it's documented.

Reply:
Update on panicking: although I still think it's not crazy to panic in this case, it turns out there isn't a convenient way to test it, because `&mut T` does not implement [`UnwindSafe`](https://doc.rust-lang.org/std/panic/trait.UnwindSafe.html). As I just learned about unwind safety today, I don't know whether the scoped operation structs can legitimately implement `UnwindSafe`.

Reply:
i think i can test it like this? 507fe6cbc850c7c0e8984341df48c0ada6de24c4

i ran into to some weird type inference issue, where if i didn't have the `assert_eq` on the line where the panic is expected, rust would complain about `a` being an unknown size at compile time, breaking the `Sized` constraint. i don't totally get what's going on, any ideas?